homeassistant/components/konnected/config_flow.py

```diff
@@ -435,123 +339,278 @@ def pins_to_zones(config):
-            if self.device_id == entry.data["id"]:
-                _LOGGER.info("Konnected host changed - creating replacement cfg entry")
-                config = copy.deepcopy(entry.data)
-                config[CONF_HOST] = self.host
-                config[CONF_PORT] = self.port
-                return await self.async_create_or_update_entry(config)
+            self._abort_if_unique_id_configured()
+            return self.async_show_form(
+                step_id="confirm",
+                description_placeholders={
+                    "model": KONN_PANEL_MODEL_NAMES[self.data[CONF_MODEL]],
+                    "host": self.data[CONF_HOST],
+                    "port": self.data[CONF_PORT],
+                },
+            )
+
+        # Attach default options and create entry
+        self.data[CONF_DEFAULT_OPTIONS] = self.options
+        return self.async_create_entry(
+            title=KONN_PANEL_MODEL_NAMES[self.data[CONF_MODEL]], data=self.data,
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Return the Options Flow."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a option flow for a Konnected Panel."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Initialize options flow."""
+        self.entry = config_entry
+        self.model = self.entry.data[CONF_MODEL]
+        self.current_opt = self.entry.options or self.entry.data[CONF_DEFAULT_OPTIONS]
+
+        # as config proceeds we'll build up new options and then replace what's in the config entry
+        self.new_opt = {CONF_IO: {}}
+        self.status = {}
+        self.active_cfg = None
+        self.io_cfg = {}
+
+    @callback
+    def get_current_cfg(self, io_type, zone):
+        """Get the current zone config."""
+        return next(
+            (
+                cfg
+                for cfg in self.current_opt.get(io_type, [])
+                if cfg[CONF_ZONE] == zone
+            ),
+            {},
+        )
 
-        return await self.async_step_io()
+    async def async_step_init(self, user_input=None):
+        """Handle options flow."""
+        return await self.async_step_options_io()
 
-    async def async_step_io(self, user_input=None):
-        """Allow the user to configure the IO."""
+    async def async_step_options_io(self, user_input=None):
+        """Configure legacy panel IO or first half of pro IO."""
         errors = {}
+        current_io = self.current_opt.get(CONF_IO, {})
+
         if user_input is not None:
             # strip out disabled io and save for options cfg
-            self.io_cfg = {}
             for key, value in user_input.items():
                 if value != CONF_IO_DIS:
-                    self.io_cfg.update({key: value})
-            return await self.async_step_io_ext()
+                    self.new_opt[CONF_IO].update({key: value})
```

Comment:
Please set the value for the key directly when possible.
```suggestion
                    self.new_opt[CONF_IO][key] = value
```